code/script.js

```diff
@@ -30,18 +74,130 @@ const showMessage = (message, sender) => {
+  })
 }
 
-// Set up your eventlisteners here
+// Interaction 4, choose what to eat.
+// As bot, confirm choice. Depending on choice, show specific menu the user may select as a drop-down. Pass the value of this choice on.
+const mainChoice = (specificChoice) => {
+  if (specificChoice === 'pizza') {
+  showMessage(`Excellent, pizza's always a winner! Make your choice below.`, 'bot')
+  inputWrapper.innerHTML = selectPizza
+  } else if (specificChoice === 'pasta') {
+  showMessage(`Can't go wrong with a pasta! Make your choice below.`, 'bot')
+  inputWrapper.innerHTML = selectPasta
+  } else {
+  showMessage(`Fresh, clean and tasty! Make your choice below.`, 'bot')
+  inputWrapper.innerHTML = selectSalad
+  }
+  const dish = document.getElementById('select')
+  addEventListener('change', () => {
+    showMessage(`Okay. I'll go with a ${dish.value}, thanks.`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => payment(dish.value), 1000)
+  })
+}
+
+// Interaction 3, looking at the menu.
+// As bot, greet the user with their name. Then present the choices the user may select. Store choice and pass it on.
+const showMainOptions = (userName) => {
+  showMessage(
+    `Nice to meet you, ${userName}. Which of our lovely main options would you like to choose?`,
+    'bot')
+  inputWrapper.innerHTML = buttonSelectMain
+  document
+  .getElementById('pizza')
+  .addEventListener('click', () => {
+    showMessage(`Let's see which pizzas you have!`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => mainChoice('pizza'), 1000)
+  })
+  document
+  .getElementById('pasta')
+  .addEventListener('click', () => {
+    showMessage(`I'd love me some pasta!`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => mainChoice('pasta'), 1000)
+  })
+  document
+  .getElementById('salad')
+  .addEventListener('click', () => {
+    showMessage(`Today I'll go with a salad.`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => mainChoice('salad'), 1000)
+  })
+}
 
-// When website loaded, chatbot asks first question.
-// normally we would invoke a function like this:
-// greeting()
-// But if we want to add a little delay to it, we can wrap it in a setTimeout:
-// setTimeout(functionName, timeToWaitInMilliSeconds)
-// This means the greeting function will be called one second after the website is loaded.
-setTimeout(greeting, 1000)
+// Interaction 2, exchange names.
+// Use the stored name, and as user reply with name. Pass stored name on.
+const handleNameInput = (event) => {
+  event.preventDefault()
+  const name = nameInput.value
+  showMessage(`My name is ${nameInput.value}.`, 'user')
+  nameInput.value = ''
+  setTimeout(() => showMainOptions(name), 1000)
+}
+
+// Interaction 1, initial greeting.
+// Ask user for name, and store name. Wait 1 second before greeting user.
+const greeting = () => {
+  showMessage(`Hello and welcome to Ziko's joint. What's your name?`, 'bot')
```

Comment:
@HIPPIEKICK Will this work when I have a single quote in the sentence itself, such as "Ziko`'`s or "Let`'`s"? VS Code underlines it, expecting commas.
I've changed to `'` where applicable (v1.1) as per your suggestion, otherwise I've left the ` for these situations.

Reply:
You're absolutely right in leaving it this way 👍